Fixed simulator error alerts, made cursor always hide, updated readme

jonchui opened the pull request:
Hey loren!

Huge fan of your work for the iOS community - thanks for doing tweetie, letterpress, pull to refresh & all the other cool stuff you've done.

Just fixed a few things in SimFinger:
- Removed cursor from ever being shown (before, it was shown on the first click)
- Removed annoying popup, because of attempt to open simulator again.
- Added steps on how to use
- Updated readme

jonchui committed:
Fixed bug where cursor shows up after first click. Now, cursor stays hidden.
Changed hideThatCursor into a c function, to be called within the tap callback

jonchui committed:
removing trying to launch simulator again, which causes errors

jonchui committed:
Added SimFinger.app file if anyone wants to download it directly

jonchui committed:
Added SimFingerApp.zip

jonchui committed:
updated README with:
How to use:
1) Launch iOS Simulator
2) Make sure the simulator's scale (Window | Scale) fits SimFinger. For retina devices, scale @ 50%. For non-retina, just use 100%
3) Launch SimFinger

Fixes
- Removed cursor from ever being shown (before, it was shown on the first click)
- Removed annoying popup, because of attempt to open simulator again.
- Added steps on how to use

jonchui committed:
Update and rename README to README.markdown

jonchui committed:
Update README.markdown

jonchui committed:
Update README.markdown

jonchui committed:
Update README.markdown

lorenbrichter commented:
Looks awesome, thanks! Rather than comment out the troublesome code, just delete it (assuming it still works, even if the simulator isn't already running?)

jonchui committed:
Added code to check if simulator not running & launch it.
TODO: still need to loop it back into rest of code path, which assumes simulator runnign

jonchui committed:
Added hack on delegate to reposition the window after a delay in app delegate
seems to work

jonchui committed:
moved the hack into the place where we first check for the simulator

jonchui committed:
added comment to hacky thing

jonchui commented:
I've fixed it so that it can launch the simulator (if it's not launched).

Okay, so tested the old code & it seems it did not launch simulator either (since it's looking for the simulator from the running applications).
- refactored out a "runningSimulatorApplication" function that returns nil if the simulator is not running
- do checks & launch the simulator from a constant, which is the bundle id "com.apple.iphonesimulator", rather than the name "iOS Simulator". Seems more future regression-safe?
- use blocks/GCD to asynchronously wait for simulator to launch & not block the main thread
- added a one-line hack to reposition the window after a 2.0 second delay, so that the newly launched simulator can gets repositioned (I don't like this, but don't have time to find a better solution)

jonchui committed:
updated .gitignore with .xcworkspace files

lorenbrichter commented:
Awesome.
